### ISSUE TITLE
SimpleFunctionRegistry.fluxifyInputIfNecessary can mistakenly recognize avro bytes as json string 

### ISSUE THREAD

**Issue by alexei-valyaev** [spring-cloud/spring-cloud-function#1031]
**tldr**
could you please backport [fix](https://github.com/spring-cloud/spring-cloud-function/commit/bbac4b198eb5775a08a36ffd45b0b52ead4e41d8) to 3.2.x and 3.1.x ?

**problem**
my application consumes avro messages from kafka via `Consumer<Message<byte[]>>` input consumer function. recently i had unexpected error on one specific message

```
Caused by: java.io.CharConversionException: Invalid UTF-32 character 0x5b01b2fb (above 0x0010ffff) at char #1, byte #7)
	at com.fasterxml.jackson.core.io.UTF32Reader.reportInvalid(UTF32Reader.java:195)
	at com.fasterxml.jackson.core.io.UTF32Reader.read(UTF32Reader.java:158)
	at com.fasterxml.jackson.core.json.ReaderBasedJsonParser._loadMore(ReaderBasedJsonParser.java:250)
	at com.fasterxml.jackson.core.json.ReaderBasedJsonParser._skipWSOrEnd(ReaderBasedJsonParser.java:2384)
	at com.fasterxml.jackson.core.json.ReaderBasedJsonParser.nextToken(ReaderBasedJsonParser.java:672)
	at com.fasterxml.jackson.databind.ObjectMapper._initForReading(ObjectMapper.java:4664)
	at com.fasterxml.jackson.databind.ObjectMapper._readMapAndClose(ObjectMapper.java:4513)
	at com.fasterxml.jackson.databind.ObjectMapper.readValue(ObjectMapper.java:3563)
	at org.springframework.cloud.function.json.JacksonMapper.doFromJson(JacksonMapper.java:55)
	at org.springframework.cloud.function.json.JsonMapper.fromJson(JsonMapper.java:94)
	at org.springframework.cloud.function.context.catalog.SimpleFunctionRegistry$FunctionInvocationWrapper.fluxifyInputIfNecessary(SimpleFunctionRegistry.java:796)
	at org.springframework.cloud.function.context.catalog.SimpleFunctionRegistry$FunctionInvocationWrapper.doApply(SimpleFunctionRegistry.java:693)
	at org.springframework.cloud.function.context.catalog.SimpleFunctionRegistry$FunctionInvocationWrapper.apply(SimpleFunctionRegistry.java:550)
	at org.springframework.cloud.stream.function.PartitionAwareFunctionWrapper.apply(PartitionAwareFunctionWrapper.java:84)
	at org.springframework.cloud.stream.function.FunctionConfiguration$FunctionWrapper.apply(FunctionConfiguration.java:749)
	at org.springframework.cloud.stream.function.FunctionConfiguration$FunctionToDestinationBinder$1.handleMessageInternal(FunctionConfiguration.java:581)
	at org.springframework.integration.handler.AbstractMessageHandler.handleMessage(AbstractMessageHandler.java:56)
```
investigation showed that `SimpleFunctionRegistry.fluxifyInputIfNecessary` tries to convert this avro bytes to json for some reason. After detailed examination of the message it became clear that it looks like json string for `JsonMapper.isJsonStringRepresentsCollection` check

here is the test for `SimpleFunctionRegistryTests`
```
@Test
public void bytesConsumerTest() {
	SimpleFunctionRegistry catalog = new SimpleFunctionRegistry(
		this.conversionService,
		this.messageConverter,
		new JacksonMapper(new ObjectMapper())
	);

	class BytesConsumer implements Consumer<Message<byte[]>> {
		@Override
		public void accept(Message<byte[]> msg) {
			System.out.println(new String(msg.getPayload(), StandardCharsets.UTF_8));
		}
	}

	FunctionRegistration<BytesConsumer> registration = new FunctionRegistration<>(
		new BytesConsumer(), "bytesConsumer").type(FunctionType.of(BytesConsumer.class));
	catalog.register(registration);

	Message<byte[]> avroValueMessage = MessageBuilder
		.withPayload(hexStringToByteArray("000000035b_this_part_doesnt_matter_it_could_be_part_of_valid_avro_msg_5d"))
		.build();

	FunctionInvocationWrapper lookedUpFunction = catalog.lookup("bytesConsumer");
	lookedUpFunction.apply(avroValueMessage);
}
```
important part is the payload '00000003**5b**...**5d**',
where '00000003' is trimmed, 5b == '[' and 5d == ']', so [this check](https://github.com/spring-cloud/spring-cloud-function/blob/3.1.x/spring-cloud-function-context/src/main/java/org/springframework/cloud/function/context/catalog/SimpleFunctionRegistry.java#L793) considers such bytes as json string

as I can see you already made the fix in main, is it possible to backport it to 3.1.x and 3.2.x?
thank you in advance!

**Comment by olegz** [comment]
Pushed to 3.2 - https://github.com/spring-cloud/spring-cloud-function/commits/3.2.x
We're not going to back port i to 3.1 as it's almost at the EOL

**Comment by alexei-valyaev** [comment]
thanks anyway
